internal/bminventory/inventory.go

```diff
@@ -2037,7 +2044,8 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 		updates["ingress_vip"] = *params.ClusterUpdateParams.IngressVip
 		ingressVip = *params.ClusterUpdateParams.IngressVip
 	}
-	if params.ClusterUpdateParams.MachineNetworkCidr != nil {
+	if params.ClusterUpdateParams.MachineNetworks != nil &&
+		len(params.ClusterUpdateParams.MachineNetworks) > 0 {
```

Comment:
I wonder whether this could be simplified to
```
if len(params.ClusterUpdateParams.MachineNetworks) > 0 {
```
because I cannot quickly craft a scenario in which MachineNetworks would not exist

Reply:
Actually it should be simplified to 
```
if params.ClusterUpdateParams.MachineNetworks != nil {
```
since we would want to support updating to an empty array.